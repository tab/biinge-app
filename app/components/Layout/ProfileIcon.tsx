import React from "react"
import { useUser } from "@realm/react"

import Avatar from "components/ui/Avatar"
import { tabBarStyles } from "styles"

const ProfileIconComponent = () => {
  const user = useUser()

  return (
    <Avatar
      style={tabBarStyles.avatar}
      email={user?.profile?.email || user.id}
    />
  )
}

export default ProfileIconComponent
