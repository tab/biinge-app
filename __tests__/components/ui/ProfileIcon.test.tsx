import React from "react"
import "react-native"
import { render } from "@testing-library/react-native"

import { it } from "@jest/globals"

import ProfileIcon from "components/ui/ProfileIcon"

jest.mock("@realm/react", () => ({
  useUser: () => ({
    email: "test@local",
  }),
}))

describe("ProfileIcon", () => {
  it("renders correctly", () => {
    const tree = render(<ProfileIcon />)
    expect(tree).toMatchSnapshot()
  })
})
