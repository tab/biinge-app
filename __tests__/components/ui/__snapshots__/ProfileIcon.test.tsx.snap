// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileIcon renders correctly 1`] = `
<View
  style={
    [
      {
        "overflow": "hidden",
      },
      {
        "height": 26,
        "width": 26,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "borderRadius": 50,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="cover"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;
